### PR TITLE
Simplified `bevy_color` `Srgba` hex string parsing

### PR DESCRIPTION
**Pull request by afonsolage**
# Objective

- Simplify `Srgba` hex string parsing using std hex parsing functions and removing loops in favor of bitwise ops.

This is a follow-up of the `bevy_color` upstream PR review: https://github.com/bevyengine/bevy/pull/12013#discussion_r1497408114

## Solution

- Reworked `Srgba::hex` to use  `from_str_radix` and some bitwise ops;

### CODE REVIEW

**Review by bushrat011899 — APPROVED** [review]
Nice change, might as well tidy these kinds of functions while we're making a new crate!

File: crates/bevy_color/src/srgba.rs [hunk 1]
```diff
@@ -83,25 +83,32 @@ impl Srgba {
...
+                    g << 4 | g,
+                    b << 4 | b,
+                    a << 4 | a,
+                ))
             }
             // RRGGBB
-            [r1, r2, g1, g2, b1, b2] => {
-                let [r, g, b, ..] = decode_hex([r1, r2, g1, g2, b1, b2])?;
+            6 => {
+                let [r, g, b, _] = (u32::from_str_radix(hex, 16)? << 8).to_be_bytes();
```
Comment (rparrett) [inline crates/bevy_color/src/srgba.rs:106]:
```suggestion
                let [_, r, g, b] = u32::from_str_radix(hex, 16)?.to_be_bytes();
```

**Review by rparrett — COMMENTED** [review]
Would be nice to add some test cases for `RGBA` and `RRGGBBAA` variants where the colors have different values.

```
assert_eq!(Srgba::hex("11223344"), Ok(Srgba::rgba_u8(17, 34, 51, 68)));
assert_eq!(Srgba::hex("1234"), Ok(Srgba::rgba_u8(17, 34, 51, 68)));
```

**Comment by afonsolage** [comment]
> Would be nice to add some test cases for `RGBA` and `RRGGBBAA` variants where the colors have different values.
> 
> ```rust
> assert_eq!(Srgba::hex("11223344"), Ok(Srgba::rgba_u8(17, 34, 51, 68)));
> assert_eq!(Srgba::hex("1234"), Ok(Srgba::rgba_u8(17, 34, 51, 68)));
> ```

Added more tests to make sure RGBA is right.

**Review by rparrett — APPROVED** [review]
Less code, more tests, and significantly faster than the old implementation.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**